packages/playwright-core/src/server/fetch.ts

```diff
@@ -204,15 +227,43 @@ export abstract class APIRequestContext extends SdkObject {
 
   private async _sendRequest(progress: Progress, url: URL, options: https.RequestOptions & { maxRedirects: number, deadline: number }, postData?: Buffer): Promise<types.APIResponse>{
     await this._updateRequestCookieHeader(url, options);
+
+    const requestCookies = (options.headers!['cookie'] as (string | undefined))?.split(';').map(p => {
+      const [name, value] = p.split('=').map(v => v.trim());
+      return { name, value };
+    }) || [];
+    const requestEvent: APIRequestEvent = {
+      url,
+      method: options.method!,
+      headers: options.headers as { [name: string]: string },
+      cookies: requestCookies,
+      postData
+    };
+    this.emit(APIRequestContext.Events.Request, requestEvent);
+
     return new Promise<types.APIResponse>((fulfill, reject) => {
       const requestConstructor: ((url: URL, options: http.RequestOptions, callback?: (res: http.IncomingMessage) => void) => http.ClientRequest)
         = (url.protocol === 'https:' ? https : http).request;
       const request = requestConstructor(url, options, async response => {
+        const notifyRequestFinished = (body?: Buffer) => {
+          const requestFinishedEvent: APIRequestFinishedEvent = {
+            requestEvent,
+            statusCode: -1,
```

Comment:
These are immediately overwritten by corresponding `...response`, we need to define them though as they are nullable in `IncomingMessage` type definition. Changed the value to `0` to avoid questions.